diff --git "..\\cores\\ddl\\library\\src\\hc32f460_pwc.c" "..\\cores\\ddl\\library\\src\\hc32f460_pwc.c"
index 8dae564..1c9dba4 100644
--- "..\\cores\\ddl\\library\\src\\hc32f460_pwc.c"
+++ "..\\cores\\ddl\\library\\src\\hc32f460_pwc.c"
@@ -129,8 +129,8 @@
 
 /*! Parameter validity check for flash mode while stop mode mode. */
 #define IS_PWC_STOP_MODE_FLASH(x)                                              \
-(   ((x) == Wait)                       ||                                     \
-    ((x) == NotWait))
+(   ((x) == FlashWait)                       ||                                \
+    ((x) == FlashNotWait))
 
 /*! Parameter validity check for wake_up timer over flag. */
 #define IS_PWC_WKTMOVER_FLAG(flag)                                             \
