diff --git "..\\cores\\ddl\\startup\\startup_hc32f460.S" "..\\cores\\ddl\\startup\\startup_hc32f460.S"
index 437b1b9..b90528d 100644
--- "..\\cores\\ddl\\startup\\startup_hc32f460.S"
+++ "..\\cores\\ddl\\startup\\startup_hc32f460.S"
@@ -17,6 +17,13 @@
 /*
 ;//-------- <<< Use Configuration Wizard in Context Menu >>> ------------------
 */
+#ifndef DDL_STACK_SIZE
+#define DDL_STACK_SIZE 0x00000400
+#endif
+
+#ifndef DDL_HEAP_SIZE
+#define DDL_HEAP_SIZE 0x00000C00
+#endif

                 .syntax     unified
                 .arch       armv7e-m
@@ -29,7 +36,7 @@
 ;  <o> Stack Size (in Bytes) <0x0-0xFFFFFFFF:8>
 ;</h>
 */
-                .equ        Stack_Size, 0x00000400
+                .equ        Stack_Size, DDL_STACK_SIZE

                 .section    .stack
                 .align      3
@@ -47,7 +54,7 @@ __StackTop:
 ;  <o> Heap Size (in Bytes) <0x0-0xFFFFFFFF:8>
 ;</h>
 */
-                .equ        Heap_Size, 0x00000C00
+                .equ        Heap_Size, DDL_HEAP_SIZE

                 .if         Heap_Size != 0                     /* Heap is provided */
                 .section    .heap
@@ -347,6 +354,8 @@ SetSRAM3Wait:

                 /* Call the clock system initialization function. */
                 bl          SystemInit
+                /* call __libc_init_array before calling main */
+                bl          __libc_init_array
                 /* Call the application's entry point. */
                 bl          main
                 bx          lr